crates/driver/src/domain/competition/order/fees.rs

```diff
@@ -15,6 +17,24 @@ pub enum FeePolicy {
         /// Cap protocol fee with a percentage of the order's volume.
         max_volume_factor: f64,
     },
+    /// A price improvement corresponds to a situation where the order is
+    /// executed at a better price than the top quote. The protocol fee in such
+    /// case is calculated from a cut of this price improvement.
+    PriceImprovement {
+        /// Factor of price improvement the protocol charges as a fee.
+        /// Price improvement is the difference between executed price and the
+        /// best quote
```

Comment:
`Price improvement is the difference between executed price and the best quote or limit price, whichever is better for the user.`